nucliadb/nucliadb/writer/tus/pg.py

```diff
@@ -0,0 +1,129 @@
+from nucliadb_utils.storages import CHUNK_SIZE
+from nucliadb_utils.storages.pg import PostgresFileDataLayer
+
+
+class PGFileStorageManager(FileStorageManager):
+    _handler = None
+    storage: PGBlobStore
+    chunk_size = CHUNK_SIZE
+
+    async def start(self, dm: FileDataMangaer, path: str, kbid: str):
+        bucket = self.storage.get_bucket_name(kbid)
+
+        async with self.storage.pool.acquire() as conn:
+            async with conn.transaction():
+                dl = PostgresFileDataLayer(conn)
+                if path is not None:
+                    await self.delete_upload(kbid, path)
+
+                await dl.create_file(
+                    kb_id=bucket,
+                    file_id=path,
+                    filename=dm.filename,
+                    size=dm.size,
+                    content_type=dm.content_type,
+                )
+
+        await dm.update(upload_file_id=path, path=path, bucket=bucket)
+
+    async def iter_data(self, uri, kbid: str, headers=None):
+        bucket = self.storage.get_bucket_name(kbid)
+
+        async with self.storage.pool.acquire() as conn:
+            dl = PostgresFileDataLayer(conn)
+            async for chunk in dl.iterate_chunks(bucket, uri):
+                yield chunk["data"]
+
+    async def read_range(
+        self, uri: str, kbid: str, start: int, end: int
+    ) -> AsyncIterator[bytes]:
+        """
+        Iterate through ranges of data
+        """
+        bucket = self.storage.get_bucket_name(kbid)
+
+        async with self.storage.pool.acquire() as conn:
+            dl = PostgresFileDataLayer(conn)
+            file_info = await dl.get_file_info(kbid, uri)
+            if file_info is None:
+                raise CloudFileNotFound()
+            async for data in dl.iterate_range(
+                kb_id=bucket, file_id=uri, start=start, end=end
+            ):
+                yield data
+
+    async def append(self, dm: FileDataMangaer, iterable, offset) -> int:
+        bucket = dm.get("bucket")
+        path = dm.get("path")
+        count = 0
+        async with self.storage.pool.acquire() as conn:
+            dl = PostgresFileDataLayer(conn)
+            async for chunk in iterable:
+                await dl.append_chunk(kb_id=bucket, file_id=path, data=chunk)
+                size = len(chunk)
+                count += size
+                offset += len(chunk)
+        return count
+
+    async def finish(self, dm: FileDataMangaer):
+        path = dm.get("path")
+        await dm.finish()
+        return path
+
+    async def delete_upload(self, uri: str, kbid: str):
+        async with self.storage.pool.acquire() as conn:
+            async with conn.transaction():
+                dl = PostgresFileDataLayer(conn)
+                await dl.delete_file(kbid, uri)
+
+
+class PGBlobStore(BlobStore):
+    def __init__(self, dsn: str):
+        self.dsn = dsn
+        self.source = CloudFile.POSTGRES
+        self._lock = asyncio.Lock()
```

Comment:
why is this lock needed?

Reply:
It's probably not necessary and the lock should be at another level